File: calling.java
/*
 * Gargi Sharma
 * Comp 6411
 */
public  class calling {

	public static void masterProcess(String text,String Sender, String Receiver, long timer)  {	
		String replyTime = ("["+timer+"]");
		try {
			long randomTime=1 + (long)(Math.random() * 100);
			Thread.sleep(randomTime);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
		if(text=="introCase"){
			long time=System.currentTimeMillis();
			String introTime = ("["+time+"]");			
			String fetchedData= Sender+" received intro message from "+Receiver+ introTime;
			exchange.getResult(fetchedData,Sender,Receiver,1,time);
		}
		if(text=="replyCase"){
			String fetchedData= Receiver+" received reply message from "+Sender+ replyTime;
			exchange.getResult(fetchedData,Sender,Receiver,2,timer);
		}
	}

}

File: calling.erl
% Gargi sharma
% Comp 6411
% Calling module

-module(calling).
-export([masterProcess/1]).

%Master Process
masterProcess(SendingData)->
    receive
    	{defaultCase, ReceivedData}->
			whereis(ReceivedData) ! {introCase, SendingData},
			masterProcess(SendingData);	

		{introCase, ReceivedData}->
			timer:sleep(rand:uniform(100)),
			Time = exchange:getTime(),
			whereis (serverModule) ! {SendingData,"received intro message from ", ReceivedData,Time},
			whereis (SendingData) ! {replyCase,ReceivedData,Time},
			masterProcess(SendingData);
		
        {replyCase,ReceivedData,Time}->
			whereis (serverModule) ! {ReceivedData, "received reply message from ",SendingData,Time},
			masterProcess(SendingData)
		
		after 1000->	
			io:fwrite("Process ~p has received no calls for 1 second, ending...~n",[SendingData])			
    end.